utils/sysUtils/httpclient.go

```diff
@@ -2,10 +2,48 @@
 package sysUtils
 
 import (
+	"context"
 	"net/http"
+	"sync"
+	"time"
 )
 
 // HTTPClient interface
 type HTTPClientI interface {
 	Do(req *http.Request) (*http.Response, error)
 }
+
+type RecycledHTTPClient struct {
+	client *http.Client
+	lock   sync.Mutex
+}
+
+func NewRecycledHTTPClient(ctx context.Context, clientFunc func() *http.Client, ttl time.Duration) *RecycledHTTPClient {
+	recycledHTTPClient := &RecycledHTTPClient{
+		client: clientFunc(),
+	}
+
+	go func() {
+		ticker := time.NewTicker(ttl)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				recycledHTTPClient.lock.Lock()
+				recycledHTTPClient.client = clientFunc()
```

Comment:
We need to run :

```suggestion
                 recycledHTTPClient.client.CloseIdleConnections()
				recycledHTTPClient.client = clientFunc()
```
https://pkg.go.dev/net/http#Client.CloseIdleConnections